DisguiseUnityRenderStream/Runtime/DisguiseRenderStream.cs

```diff
@@ -432,6 +396,75 @@ protected void AwaitFrame()
             DisguiseFramerateManager.Update();
         }
 
+        protected void FinishFrameRendering()
+        {
+            NativeRenderingPlugin.OnFrameEnd();
+            Texture2DPool.Instance.OnFrameEnd();
+        }
+
+        // Updates the RenderTextures assigned to image parameters on the render thread to avoid stalling the main thread
+        protected void OnBeginContextRendering(ScriptableRenderContext context, List<Camera> cameras)
+        {
+            if (!HasNewFrameData)
+                return;
+            
+            if (LatestFrameData.scene >= m_Schema.scenes.Length)
+                return;
+        
+            var spec = m_Schema.scenes[LatestFrameData.scene];
+            var images = m_SceneFields[LatestFrameData.scene].images;
+            if (images == null)
+                return;
+
+            // Only run once per frame for the main render context
+            if (cameras.Any(x => x.cameraType != CameraType.Game))
+                return;
+            
+            var nImageParameters = spec.parameters.Count(t => t.type == RemoteParameterType.RS_PARAMETER_IMAGE);
+
+            using var imageData = new NativeArray<ImageFrameData>(nImageParameters, Allocator.Temp);
+            if (PluginEntry.instance.GetFrameImageData(spec.hash, imageData.AsSpan()) != RS_ERROR.RS_ERROR_SUCCESS)
+                return;
+
+            var i = 0;
+            foreach (var field in images)
+            {
+                if (field.GetValue() is RenderTexture renderTexture)
+                {
+                    var texture = Texture2DPool.Instance.Get(new Texture2DDescriptor() {
+                        Width = (int)imageData[i].width,
+                        Height = (int)imageData[i].height,
+                        Format = PluginEntry.ToTextureFormat(imageData[i].format),
+                        Linear = true
+                    });
+                    
+                    var cmd = CommandBufferPool.Get($"Receiving Disguise Image Parameter '{field.info.Name}'");
+                    
+                    NativeRenderingPlugin.GetFrameImageData data = new NativeRenderingPlugin.GetFrameImageData()
+                    {
+                        m_rs_getFrameImage = PluginEntry.instance.rs_getFrameImage_ptr,
+                        m_ImageId = imageData[i].imageId,
+                        m_Texture = texture.GetNativeTexturePtr()
+                    };
+                    
+                    cmd.IssuePluginEventAndData(
+                        NativeRenderingPlugin.GetRenderEventCallback(),
+                        (int)NativeRenderingPlugin.EventID.GetFrameImage,
+                        NativeRenderingPlugin.GetFrameImageDataPool.Pin(data));
+                    cmd.IncrementUpdateCount(texture);
+                    
+                    cmd.Blit(texture, renderTexture, new Vector2(1.0f, -1.0f), new Vector2(0.0f, 1.0f));
+                    cmd.IncrementUpdateCount(renderTexture);
```

Comment:
Curious: There's no need to "decrement" the count?

Reply:
It looks like a notification mechanism so it doesn't have a decrement counterpart: https://tryfinally.dev/increment-update-count. I guess in the editor it would trigger a texture inspector window refresh for ex.

Reply:
Ah, I misunderstood what the function does. I thought it was some kind of reference-counting mechanism.